lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -4023,9 +4023,10 @@ def as_cardan_angles(self):
         The inverse of `from_cardan_angles()`.
         Note that the angles returned are in radians, not degrees.
         """
-        qw = self.scalar
-        qx, qy, qz = self.vector[..., :]
+        n = np.sqrt(self.norm)
+        qw = self.scalar / n
+        qx, qy, qz = self.vector[..., :] / n
```

Comment:
This replaced a single division by 4 of them, and added an extra sqrt to boot. And it made it harder to understand what the range of the arcsin argument is... Improvement?

Reply:
As the quaternion is not normed upon initialization, this step is necessary to have the trig functions following to be valid. In its current use case the quaternions are always unit length, but there is no guarantee of that. I think the alternative to norming here would be to throw an error if this method is called with a non-unit-norm quaternion.

Reply:
The step is not necessary, the routine is not sensitive to the magnitude of the quaternion, the way it was before (only  ratios occur in the arcsin arguments); the argument of the arcsin is always in range [-1,+1], it seems perfectly valid. Please revert (or provide a counter example).

Reply:
As a counterexample, `qw, qx, qy, qz = 1, 0, 1, 0`. Before the change this would result in elev = 30 deg, and after it results in 45 deg as expected. The sqrt is necessary for norming.

Reply:
Actually, it wasn't immediately obvious to me but I see now that you put in simplified forms that baked the normalization into the trig functions. I can revert this portion of the change.

Reply:
> Actually, it wasn't immediately obvious to me but I see now that you put in simplified forms that baked the normalization into the trig functions. 

Should we put a note to clarify/make it more obvious, for our future audiences? 

> I can revert this portion of the change.

Yes please :)

Reply:
Done for both